### PR TITLE
feat(types): add JSONPatchFormatter

### PR DESCRIPTION
**Pull request by marcolink**
Add type definition for JSON Patch formatter matching official `Operation` definition from http://jsonpatch.com

### CODE REVIEW

**Comment by marcolink** [comment]
@benjamine what do you think? are we good?

**Comment by Vinz93** [comment]
Any updates here? :)

**Comment by sghoweri** [comment]
@marcolink oh my gosh you're a lifesaver (was just dealing with the Typescript typing issues with this package) 🙏🏼

**Comment by sghoweri** [comment]
> Elegant array diffing by providing an objectHash to match array elements

Oh snap. This is precisely why we started digging into this library to begin with (array diffing combined with JSON patches). 👀

File: src/index.d.ts [hunk 1]
```diff
@@ -19,6 +19,34 @@ export interface HtmlFormatter extends Formatter {
     hideUnchanged(node?: Element | null, delay?: number): void;
 }
 
+type AddOperation = {
+  opp: string;
```
Comment (Methuselah96) [inline src/index.d.ts]: Seems like these should be `op` and not `opp`?

File: src/index.d.ts [hunk 2]
```diff
@@ -19,6 +19,34 @@ export interface HtmlFormatter extends Formatter {
...
+}
+
+type MoveOperation = {
+  opp: string;
+  path: string;
+}
+
+export type Operation = AddOperation | RemoveOperation | ReplaceOperation | MoveOperation
+
+export interface JSONPatchFormatter {
```
Comment (Methuselah96) [inline src/index.d.ts:46]: Ideally this would inherit from `Formatter`, but we can handle that in a future PR.

**Review by Methuselah96 — CHANGES_REQUESTED** [review]
Apologies for the delay review. Don't feel you need to update this PR, we plan on converting the codebase to TypeScript soon anyway, so these changes will likely get overridden, but feel free to update the PR with the requested changes and we can get it merged.

**Pull request closed without merging**